### PR TITLE
Release v4.3.0

### PR DESCRIPTION
**Pull request by gregyu**

### CODE REVIEW

File: knockoff/sdk/table.py [hunk 1]
```diff
@@ -163,26 +263,27 @@ def copy_factory(self, columns=None, rename=None):
...
+
         return factory
 
-    def build_record(self):
-        """TODO: should this be public?"""
+    def _build_record(self):
+        """
+        TODO: should this be public?
+              we can make this private and
+              use apply the column changes later
```
Comment (abdulhuq811) [inline knockoff/sdk/table.py:278]: Is there a typo here?

Comment (gregyu) [reply knockoff/sdk/table.py:278]: yes, will remove that comment as _build_record was made private by convention

File: knockoff/utilities/io.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+           url,
+           parallelize=True,
+           chunksize=1000,
+           n_jobs=-1,
+           **kwargs):
+    logger.info("Populating table: {}".format(table))
+    # TODO: better default for more effect parallelization?
+    nrows = df.shape[0]
+    if parallelize and nrows > chunksize:
+        Parallel(n_jobs=n_jobs)(
```
Comment (abdulhuq811) [inline knockoff/utilities/io.py:32]: Does the table already exists when the inserts are happening?
I did run into an edge case where the table did not exist and multiple threads/processes tried to create it. I had to use lock if creating a new table and writing to it.

**Review by abdulhuq811 — APPROVED** [review]

**Comment by abdulhuq811** [comment]
LGTM

File: knockoff/utilities/io.py (hunk 2 above)
Comment (gregyu) [reply knockoff/utilities/io.py:32]: No assumptions are made here as to whether the tables exists or not, but I imagine most use-cases will be populating pre-existing tables (potentially through the TempDatabaseService)

**Pull request merged**